pygmt/helpers/utils.py

```diff
@@ -213,6 +217,10 @@ def build_arg_string(kwargs):
                 # remove any spaces in PROJ4 string
                 _value = str(kwargs[key]).replace(" ", "")
             gmt_args.append(rf"-{key}{_value}")
+    if infile:
+        gmt_args.append(str(infile))
+    if outfile:
+        gmt_args.append("->" + str(outfile))
     return " ".join(sorted(gmt_args))
```

Comment:
I don't like how the arguments are sorted. For example, currently, the returned argument string may look like:
```
->outfile -A0 -Bafg -Ctopo -Vd infile
```
but I expect to see a more readable argument string like:
```
infile -A0 -Bafg -Ctopo -Vd ->outfile
```
which is more friendly when debugging.

I can rewrite these lines to 
```
gmt_args = sorted(gmt_args)
if infile:
    gmt_args = [str(infile)] + gmt_args
if outfile:
    gmt_args = gmt_args.append("->" + str(outfile))
return " ".join(gmt_args)
```
but I'm wondering if someone can give a better and cleaner solution.

Reply:
Something like this? Based on https://stackoverflow.com/questions/3752240/join-string-and-none-string-using-optional-delimiter/3752317#3752317
```suggestion
    return " ".join(filter(None, [infile, sorted(gmt_args), outfile]))
```